pkg/sql/plan2/build.go

```diff
@@ -35,6 +34,18 @@ func buildPlan(ctx CompilerContext, stmt tree.Statement) (*Query, error) {
 	return query, nil
 }
 
+func BuildPlan2(ctx CompilerContext, stmt tree.Statement) (*plan.Query, error) {
```

Comment:
you can move /plan2/explain/buildexplain_test.go to /plan2/
then you can remove BuildPlan2()  and NewMockOptimizer2（）